Resources/doc/reference/configuration/user-context.rst

```diff
@@ -139,26 +139,18 @@ or you will end up with mixed up caches.
 ~~~~~~~~~~~~~~~~~~
 
 The logout handler will invalidate any cached user hashes when the user logs
-out.
+out. This behavior applies to all stateful firewalls.
```

Comment:
i would prefer a `.. note::` block with this plus the extra text you add below. something like: "The logout handler is active on all firewalls. If your application has multiple firewalls with different user contexts, you need to create your own custom invalidation handler. Be aware that Symfony's ``LogoutSuccessHandler`` places the ``SessionLogoutHandler`` before any configured logout handlers."